File: apps/web/src/lib/audit.ts
import { api } from "@/lib/api";

export interface AuditEntry {
  id: number;
  actor_user_id: number | null;
  actor_label: string;
  action: string;
  entity_type: string;
  entity_id: number | null;
  summary: string | null;
  meta: Record<string, unknown> | null;
  created_at: string;
}

export async function fetchAuditLog(params?: {
  entity_type?: string;
  action?: string;
  limit?: number;
}): Promise<AuditEntry[]> {
  return (await api.get<AuditEntry[]>("/audit-log", { params })).data;
}

File: apps/api/app/schemas/audit.py
"""Schemas for the audit-log viewer."""

from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel


class AuditRead(BaseModel):
    id: int
    actor_user_id: int | None
    actor_label: str
    action: str
    entity_type: str
    entity_id: int | None
    summary: str | None
    meta: dict[str, Any] | None
    created_at: datetime
